test/csuite/wt8246_compact_rts_data_correctness/main.c

```diff
@@ -0,0 +1,378 @@
+
+    /* Kill the child process. */
+    printf("Kill child\n");
+    testutil_checksys(kill(pid, SIGKILL) != 0);
+    testutil_checksys(waitpid(pid, &status, 0) == -1);
+
+    printf("Compact process interrupted and killed...\n");
+
+    /* Copy the data to a separate folder for debugging purpose. */
+    testutil_copy_data(home);
+
+    printf("Open database and run recovery\n");
+
+    /* Open the connection which forces recovery to be run. */
+    testutil_check(wiredtiger_open(home, NULL, ENV_CONFIG_REC, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Get the stable timestamp from the stable timestamp of the last successful checkpoint. */
+    testutil_check(conn->query_timestamp(conn, ts_string, "get=stable"));
+    testutil_timestamp_parse(ts_string, &stable_ts);
+
+    /* Get the oldest timestamp from the oldest timestamp of the last successful checkpoint. */
+    testutil_check(conn->query_timestamp(conn, ts_string, "get=oldest"));
+    testutil_timestamp_parse(ts_string, &oldest_ts);
+
+    /*
+     * Verify data is visible and correct after compact operation was killed and RTS is performed in
+     * recovery. Stable timestamp is used to check the visibility of data, stable timestamp is
+     * retrieved from the connection using query_timestamp.
+     */
+    check(session, uri, value_a, 20);
+    check(session, uri, value_b, 30);
+    check(session, uri, value_b, 40);
+    check(session, uri, value_b, 50);
+
+    testutil_check(session->checkpoint(session, NULL));
+
+    testutil_check(session->close(session, NULL));
+    session = NULL;
+
+    testutil_check(conn->close(conn, NULL));
+    conn = NULL;
+    return (EXIT_SUCCESS);
+}
+
+static void
+workload_compact(const char *home, const char *table_config, const char *uri)
+{
+    FILE *fp;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    char compact_file[2048], tscfg[64];
+
+    testutil_check(wiredtiger_open(home, NULL, conn_config, &conn));
+
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    testutil_check(__wt_snprintf(tscfg, sizeof(tscfg),
+      "oldest_timestamp=%d"
+      ",stable_timestamp=%d",
+      10, 10));
+    testutil_check(conn->set_timestamp(conn, tscfg));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, uri, table_config));
+
+    populate(session, uri);
+    testutil_check(session->checkpoint(session, NULL));
+
+    /* Perform several updates. */
+    large_updates(session, uri, value_a, 20);
+    large_updates(session, uri, value_b, 30);
+    large_updates(session, uri, value_c, 40);
+    large_updates(session, uri, value_d, 50);
+
+    /* Verify data is visible and correct. */
+    check(session, uri, value_a, 20);
+    check(session, uri, value_b, 30);
+    check(session, uri, value_c, 40);
+    check(session, uri, value_d, 50);
+
+    /* Pin stable to timestamp 40. */
```

Comment:
The comment is saying stable timestamp as 40, but the setting is done with 30.

Reply:
Fixed